src/persist/src/indexed/mod.rs

```diff
@@ -143,6 +177,7 @@ pub struct Indexed<L: Log, B: Blob> {
     listeners: HashMap<Id, Vec<ListenFn<Vec<u8>, Vec<u8>>>>,
     metrics: Metrics,
     state: AppliedState,
+    in_flight_trace_compactions: HashSet<Id>,
```

Comment:
i think persist probably wants to keep a HashMap<Id, CompactTraceRequest> and only toggle its state of "do I have a request in flight" from true -> false if the response from Maintainer matches the request (to be resilient against duplicate responses). I think yeah, we can try to make use of the response if it doesn't match the request we thought we sent, but I'd rather not do so for now because the connection between indexed and maintainer is infallible and won't duplicate responses so that code wouldn't ever be used.

Reply:
Did ^